Tutorial Panel added to main menu and two game scenes

WendyFu004 opened the pull request:
Question mark on the left top corner is the help bottom used to open the tutorial page, and this button shown in main menu and both game scenes. When the tutorial page is opening, the time scale become zero, so the game is paused during the time when the tutorial is on.

First try:  Click the bottom to open the tutorial page and close the page with click again.

** Now what we have: H key is used to open and close the tutorial now! I found that we could use both arrow   keys and mouse to choose the items so I change to use the key otherwise the mouse click will mess up with spacebar.

**made an simple image for the background. Replaced the original one, and might make a cover (poster?) if time enough.

WendyFu004 committed:
Tutorial Panel added to main menu and two game scenes

Question mark on the left top corner is the help bottom used to open the tutorial page, and this button shown in main menu and both game scenes. When the tutorial page is opening, the time scale become zero, so the game is paused during the time when the tutorial is on.

First try:  Click the bottom to open the tutorial page and close the page with click again.

** Now what we have: H key is used to open and close the tutorial now! I found that we could use both arrow   keys and mouse to choose the items so I change to use the key otherwise the mouse click will mess up with spacebar.

**made an simple image for the background. Replaced the original one, and might make a cover (poster?) if time enough.

WendyFu004 committed:
Rescale new background image and made it tiled

WendyFu004 committed:
Fixed time scale bug

Aether700 commented on Assets/Scripts/UI/PauseMenu/PauseMenu.cs:
```diff
@@ -5,7 +5,7 @@
 public class PauseMenu : MonoBehaviour
 {
-    private static bool GameIsPaused = false;
+    public static bool GameIsPaused = false;
```
Please do not make this variable public and use getters/setters to access/modify the value of the variable

Aether700 reviewed (changes requested):
Very good work. There is a coding issue I would like you to fix though. I left a comment describing the issue above.

Also, small tweaks I would suggest: in the race track scene move the help indicator at the very top left corner of the screen and make sure the question mark sprite is not highlighted like a button when you hover it with your mouse because it looks like a button that can be clicked when it is not

afourneaux commented:
Absolutely gorgeous. I noticed "Rocket" is abbreviated to "Rock" which seems odd to me, but the images make it very clear.

Fantastic work, I love the look of it!

WendyFu004 committed:
Fixed typo. Script fixed and remove button UI of the question mark

Add a new script for the main menu tutorial since it don't have a paus feature. Change the bool to private.

afourneaux commented:
In the main menu, it looks like the H? icon is blocking clicks to the dumpster. Simply toggling "Raycast Target" on the QuestionMark game object should resolve this

WendyFu004 committed:
Fixed that ray casting problem. Thank you!

Aether700 commented on Assets/Scripts/Tutor/TutForMenu.cs:
```diff
@@ -0,0 +1,30 @@
+
+public class TutForMenu : MonoBehaviour
+{
+    public GameObject TutorialPanel;
```
Please make this variable private and add the serializeField attribute to it

Aether700 commented on Assets/Scripts/Tutor/Tutoral.cs:
```diff
@@ -5,6 +5,8 @@
 public class Tutoral : MonoBehaviour
 {
     public GameObject TutorialPanel;
```
Again make this variable private and add the serialize field to it

WendyFu004 committed:
Game objects changed to private. Thanks for discovering this

Aether700 reviewed (dismissed):
nice work

WendyFu004 committed:
Merge branch 'main' into Tutorial

Aether700 reviewed (changes requested):
The canyon scene crashes when loaded Please fix

WendyFu004 committed:
Fixed the bullseye crashes but still have a problem to star the game? Not sure why this happened

Aether700 commented:
The issue is probably how you merged main
I would look there to make sure that's not the problem first

afourneaux commented:
Please see #49 where the merge was corrected
